### PR TITLE
:sparkles: Managed cluster events inventory for global hub manager 

### PR DESCRIPTION
**Pull request by yanmxa**
## Summary

Provide the events inventory for managed clusters. The cluster event from the managed hub:
```yaml
apiVersion: v1
items:
- action: AvailableUnknown
  apiVersion: v1
  eventTime: "2024-05-07T12:33:27.502217Z"
  firstTimestamp: null
  involvedObject:
    apiVersion: cluster.open-cluster-management.io/v1
    kind: ManagedCluster
    name: globalhub-cluster1
    namespace: globalhub-cluster1
    resourceVersion: "747936"
    uid: 39c1aeff-0f51-48f0-a380-2b36defa3f59
  kind: Event
  lastTimestamp: null
  message: The managed cluster (globalhub-cluster1) cannot connect to the hub cluster.
  metadata:
    creationTimestamp: "2024-05-07T12:33:27Z"
    name: globalhub-cluster1.17cd34e8c8b27fdd
    namespace: globalhub-cluster1
    resourceVersion: "754506"
    uid: c3c1ea9d-183e-4e7c-bb44-537f316669a6
  reason: AvailableUnknown
  reportingComponent: registration-controller
  reportingInstance: registration-controller-cluster-manager-registration-controller-6794cf54d9-j7lgm
  source: {}
  type: Warning
- action: Available
  apiVersion: v1
  eventTime: "2024-05-07T12:34:07.319090Z"
  firstTimestamp: null
  involvedObject:
    apiVersion: cluster.open-cluster-management.io/v1
    kind: ManagedCluster
    name: globalhub-cluster1
    namespace: globalhub-cluster1
    resourceVersion: "754531"
    uid: 39c1aeff-0f51-48f0-a380-2b36defa3f59
  kind: Event
  lastTimestamp: null
  message: The managed cluster (globalhub-cluster1) is available now
  metadata:
    creationTimestamp: "2024-05-07T12:34:07Z"
    name: globalhub-cluster1.17cd34f21203e39b
    namespace: globalhub-cluster1
    resourceVersion: "754951"
    uid: d1383786-f39a-4460-ace0-7dba464ced62
  reason: Available
  reportingComponent: klusterlet-agent
  reportingInstance: klusterlet-agent-klusterlet-agent-9f7f5d4d8-644hj
  source: {}
  type: Normal
- action: AvailableUnknown
  apiVersion: v1
  eventTime: "2024-05-07T12:40:59.895652Z"
  firstTimestamp: null
  involvedObject:
    apiVersion: cluster.open-cluster-management.io/v1
    kind: ManagedCluster
    name: globalhub-cluster1
    namespace: globalhub-cluster1
    resourceVersion: "755295"
    uid: 39c1aeff-0f51-48f0-a380-2b36defa3f59
  kind: Event
  lastTimestamp: null
  message: The managed cluster (globalhub-cluster1) cannot connect to the hub cluster.
  metadata:
    creationTimestamp: "2024-05-07T12:40:59Z"
    name: globalhub-cluster1.17cd35521d72897b
    namespace: globalhub-cluster1
    resourceVersion: "758945"
    uid: bf52a104-63ae-420c-a0fb-24aaa109c552
  reason: AvailableUnknown
  reportingComponent: registration-controller
  reportingInstance: registration-controller-cluster-manager-registration-controller-6794cf54d9-j7lgm
  source: {}
  type: Warning
kind: List
metadata:
  resourceVersion: ""
```

Deprecated `source` filed in event: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#event-v125

## Related issue(s)

References # https://issues.redhat.com/browse/ACM-9676

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/database/2.tables.sql [hunk 1]
```diff
@@ -59,6 +59,20 @@ CREATE TABLE IF NOT EXISTS status.leaf_hubs (
...
 
 -- Partition tables
+CREATE TABLE IF NOT EXISTS event.managed_clusters (
+    event_namespace text NOT NULL,
+    event_name text NOT NULL,
+    cluster_id uuid NOT NULL,
+    leaf_hub_name character varying(256) NOT NULL,
+    message text,
+    reason text,
+    reporting_controller text,
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/database/2.tables.sql:69]: Do you mean `source` is not recommended? use `reporting_controller` instead?

Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/database/2.tables.sql:69]: Yes! https://kubernetes.io/docs/reference/using-api/deprecation-guide/#event-v125

**Review by clyang82 — APPROVED** [review]
LGTM

File: operator/pkg/controllers/hubofhubs/database/2.tables.sql (hunk 1 above)
Comment (clyang82) [reply operator/pkg/controllers/hubofhubs/database/2.tables.sql:69]: Do you think we need to modify the policy event table?

**Pull request merged**